src/lib.rs

```diff
@@ -26,73 +26,92 @@
 #![deny(unused_imports)]
 #![deny(missing_docs)]
 
+#![cfg_attr(all(not(test), not(feature = "std")), no_std)]
+#[cfg(any(test, feature = "std"))] pub extern crate core;
+
 extern crate bitcoin_hashes;
+#[cfg(feature = "std")]
 extern crate unicode_normalization;
+
 #[cfg(feature = "rand")]
 extern crate rand;
+#[cfg(feature = "serde")]
+pub extern crate serde;
 
+#[cfg(feature = "std")]
 use std::{error, fmt, str};
+#[cfg(feature = "std")]
 use std::borrow::Cow;
 
 use bitcoin_hashes::{sha256, Hash};
+
+#[cfg(feature = "std")]
 use unicode_normalization::UnicodeNormalization;
 
+#[macro_use]
+mod internal_macros;
 mod language;
 mod pbkdf2;
 
 pub use language::Language;
 
-/// The ideagrapic space that should be used for Japanese lists.
-#[cfg(feature = "japanese")]
-#[allow(unused)]
-const IDEOGRAPHIC_SPACE: char = '　';
+/// The maximum number of words in a mnemonic.
+const MAX_NB_WORDS: usize = 24;
 
 /// A BIP39 error.
-#[derive(Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Copy)]
 pub enum Error {
 	/// Mnemonic has a word count that is not a multiple of 6.
 	BadWordCount(usize),
 	/// Mnemonic contains an unknown word.
-	UnknownWord(String),
+	UnknownWord, //TODO(stevenroose) add back word info
```

Comment:
Do you plan to fix this? It would probably result in an API break. I guess the problem was allocating? If so can you return a slice into the original string? This would give the error type a lifetime, which is annoying, but probably ok since upstream can then decide what to do with it. If you feel fancy you could even provide an `OwnedError` type if `std` is present  which implements `From<Error>`

Reply:
I fixed this with a word index that we can always provide. Should be good enough?